app/pages/sections/catalogue-solutions/routes.js

```diff
@@ -49,19 +49,30 @@ export const catalogueSolutionsRoutes = (authProvider, addContext, sessionManage
   router.get('/:orderItemId', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
     const { orderId, orderItemId } = req.params;
     const accessToken = extractAccessToken({ req, tokenType: 'access' });
-
-    if (orderItemId !== 'newsolution') {
-      await getOrderItem({ orderId, orderItemId, accessToken });
-    }
+    const selectedPriceId = sessionManager.getFromSession({ req, key: 'selectedPriceId' });
     const selectedSolutionId = sessionManager.getFromSession({ req, key: 'selectedSolutionId' });
+    const solutionName = (await getSolution({ solutionId: selectedSolutionId, accessToken })).name;
     const selectedRecipientId = sessionManager.getFromSession({ req, key: 'selectedRecipientId' });
     const serviceRecipientName = sessionManager.getFromSession({ req, key: 'selectedRecipientName' });
+    let selectedPrice;
+    let formData;
 
-    const solutionName = (await getSolution({ solutionId: selectedSolutionId, accessToken })).name;
-    sessionManager.saveToSession({ req, key: 'solutionName', value: solutionName });
+    if (orderItemId !== 'newsolution') {
+      selectedPrice = await getOrderItem({ orderId, orderItemId, accessToken });
+      const date = selectedPrice.deliveryDate.split('-');
+      formData = {
+        'deliveryDate-year': date[0],
+        'deliveryDate-month': date[1],
+        'deliveryDate-day': date[2],
+        quantity: selectedPrice.quantity,
+        selectEstimationPeriod: selectedPrice.estimationPeriod,
+        price: selectedPrice.price,
+      };
+    } else {
+      selectedPrice = await getSelectedPrice({ selectedPriceId, accessToken });
+    }
```

Comment:
I think we can tidy this up a little. Let's add a function in the controller called something like `getFormData({ orderId, orderItemId, selectedPriceId, accessToken })` and we can move this logic there. This way this function will work out whether to get this data from ordapi or bapi